### PR TITLE
PXP-4884 added tests to make sure we can remove values (only non required values)

### PR DESCRIPTION
**Pull request by mjmartinson**
Related to https://github.com/uc-cdis/dictionaryutils/pull/59 and https://github.com/uc-cdis/data-simulator/pull/39

### New Features

- Implemented the ability to remove values from fields that are not required in existing entities (use null in JSON, empty values in TSV/CSV).

### Dependency updates
dictionaryutils to 3.1.0

### Deployment changes

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
This PR contains code that is not formatted correctly according to [`black`](https://github.com/ambv/black). Run `black` on your code before merging, or comment "wool" or "black" on this PR and I will format the code for you.

<details>
<summary>Expand the full diff to see formatting changes</summary>

```diff
--- tests/integration/datadict/submission/test_endpoints.py
+++ blackened
@@ -1030,10 +1030,11 @@
         }
     )
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     assert resp.status_code == 400, resp.data
 
+
 def test_update_to_null_invalid2(client, pg_driver, cgci_blgsp, submitter):
     headers = submitter
     data = json.dumps(
         {
             "type": "experiment",
@@ -1042,17 +1043,14 @@
         }
     )
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     assert resp.status_code == 200, resp.data
 
-    data = json.dumps(
-        {
-            "type": "null",
-        }
-    )
+    data = json.dumps({"type": "null",})
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     assert resp.status_code == 400, resp.data
+
 
 def test_update_to_null_invalid3(client, pg_driver, cgci_blgsp, submitter):
     headers = submitter
     data = json.dumps(
         {
@@ -1062,15 +1060,11 @@
         }
     )
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     assert resp.status_code == 200, resp.data
 
-    data = json.dumps(
-        {
-            "id": None,
-        }
-    )
+    data = json.dumps({"id": None,})
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     assert resp.status_code == 400, resp.data
 
 
 def test_update_to_null_valid_tsv(client, pg_driver, cgci_blgsp, submitter):
@@ -1204,10 +1198,11 @@
     headers["Content-Type"] = "text/tsv"
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     print(json.dumps(json.loads(resp.data), indent=4, sort_keys=True))
     assert resp.status_code == 400, resp.data
 
+
 def test_update_to_null_invalid_tsv2(client, pg_driver, cgci_blgsp, submitter):
     """
     Test that we can update a TSV file with null
     """
 
@@ -1263,10 +1258,11 @@
     headers["Content-Type"] = "text/tsv"
     resp = client.put(BLGSP_PATH, headers=headers, data=data)
     print(json.dumps(json.loads(resp.data), indent=4, sort_keys=True))
     assert resp.status_code == 400, resp.data
 
+
 def test_update_to_null_invalid_tsv3(client, pg_driver, cgci_blgsp, submitter):
     """
     Test that we can update a TSV file with null
     """
 
```
</details></br>

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: sheepdog/utils/transforms/__init__.py [hunk 1]
```diff
@@ -76,7 +76,8 @@ def get_links_from_row(row):
 
 def get_props_from_row(row):
     """Return a dict of key/value pairs that are props, not links."""
-    return {k: v for k, v in row.items() if "." not in k and v != ""}
+    return {k: v for k, v in row.items() if "." not in k}
+    # return {k: v for k, v in row.items() if "." not in k and v != ""}
```
Comment (paulineribeyre) [inline sheepdog/utils/transforms/__init__.py:80]: remove the commented out line

File: tests/integration/datadict/submission/test_endpoints.py [hunk 2]
```diff
@@ -1030,3 +1030,236 @@ def test_zero_decimal_float(client, pg_driver, cgci_blgsp, submitter):
...
+        "submitter_id": "BLGSP-71-06-00019",
+        "projects.id": "daa208a7-f57a-562c-a04a-7a7c77542c98",
+        "experimental_description": None,
+        "number_samples_per_experimental_group": None,
+    }
+
+    # convert to TSV (save to file)
+    file_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), "data/experiment_tmp.tsv"
+    )
```
Comment (paulineribeyre) [inline tests/integration/datadict/submission/test_endpoints.py:1171]: so in the resulting file, the `None` are empty values right?

File: sheepdog/utils/transforms/__init__.py [hunk 3]
```diff
@@ -148,7 +149,9 @@ def add_row(self, row):
         # Add properties
         props_dict = get_props_from_row(row)
         for key, value in props_dict.items():
-            if value == "null":
+            if value == "":
+                doc[key] = None
```
Comment (paulineribeyre) [inline sheepdog/utils/transforms/__init__.py:153]: please add a comment to explain what this `if` is accomplishing

File: openapi/swagger.yml [hunk 4]
```diff
@@ -342,7 +342,9 @@ paths:
     put:
       description: Create or update any valid entities specified in the request body.
         To associate an entity with an existing, already indexed data file, specify
-        the `object_id` in the body of the entity.
+        the `object_id` in the body of the entity. To remove a non-required property 
+        of a valid entity (set it to `Null`) you must set that property's value to `"null"`
```
Comment (paulineribeyre) [inline openapi/swagger.yml:346]:
```suggestion
        of an existing entity (set it to `Null`) you must set that property's value to
        `None` (in JSON submissions) or an empty value (in TSV/CSV submissions)
```

File: tests/integration/datadict/submission/test_endpoints.py (hunk 2 above)
Comment (mjmartinson) [reply tests/integration/datadict/submission/test_endpoints.py:1171]: Yes

**Pull request merged**